### ISSUE TITLE
Order of task's tags added through code

### ISSUE THREAD

**Issue by israelwei** [allegroai/clearml#335]
Hi,
I can see that it's possible to [add task's tag through code](https://allegro.ai/clearml/docs/rst/references/clearml_python_ref/task_module/task_task.html#clearml.task.Task.add_tags).
My question is- is it possible to add a feature for having a control over the order in which the tags appear in the UI?
For example, suppose we have two tags: one for dataset name and one for task type, as in the following image:
![image](https://user-images.githubusercontent.com/23454156/115144759-1b90b780-a057-11eb-9519-d7250a853c35.png)
We would like to have the tag of dataset_name to be the very left, second to the left the task type, after this another type of tag etc...

Thank you

**Comment by jkhenning** [comment]
Hi @israelwei, right now the tags are sorted by the server. Do you have any suggestion as to how this control should be provided?

**Comment by Mano3** [comment]
Hi Jake, 
Sorry for intervening, I think the best practice would be to use the order given by the user, i.e. the order in which tags were added.

**Comment by assafzam** [comment]
> Hi @israelwei, right now the tags are sorted by the server. Do you have any suggestion as to how this control should be provided?


1. by the order of adding the tags:

```
task.add_tags('first')
task.add_tags('second')
task.add_tags(['third', 'fourth'])

>>> will cause 'first', 'second', 'third', 'fourth'
```
2. option to the user to provide comparator func

**Comment by bmartinn** [comment]
I "think" this is the only issue, no?
https://github.com/allegroai/clearml/blob/8658198f8bac976e5c2766190d8c593c2308bfa1/clearml/task.py#L1042

If this is the case, it could be easily fixed (i.e. the `set` casting should be replace as it will not keep the original order).

**Comment by jkhenning** [comment]
@bmartinn not really, since the main reason for this sorting is to keep a consistent user-interface for users adding tags from the UI, otherwise experiments might be very different visually, while containing the exact same tags.
For storing consistent data-types with user-specified ordering, you can always use the hyper-parameters or user properties...

**Comment by patrickdasilva-clearml** [comment]
Hi @israelwei , we're closing this issue as stale since it's been created too long ago. If you think the issue is still relevant, please make sure the issue persists on a recent version of ClearML (ideally the latest) before re-opening the issue. We'll gladly look at it again.
